src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -406,6 +414,135 @@ class CDCL {
         },
     )
 
+    // ---- Preprocessing ---- //
+
+    /**
+     * Preprocessing is ran before the main loop of the solver,
```

Comment:
Maybe replace this docs with the direct explanation of what preprocessing is, or what this method does? ("perform the preprocessing: try to simplify the SAT problem using different techniques and heuristics")

Also, remember the inprocessing: are you going to create a separate method for it, or do you inclined to reuse `preprocessing`?